File: app/src/main/java/com/example/ibook_social_network/ConnectionsActivity.java
package com.example.ibook_social_network;

import android.content.Context;
import android.content.Intent;
import android.net.wifi.WifiManager;
import android.os.Build;
import android.os.Bundle;
import android.provider.Settings;
import android.view.View;

import androidx.activity.result.ActivityResultLauncher;
import androidx.activity.result.contract.ActivityResultContracts;
import androidx.appcompat.app.AppCompatActivity;

import java.util.Objects;

public class ConnectionsActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_connections);
        Configuration.awp(getWindow(), Objects.requireNonNull(getSupportActionBar()));
    }

    public void settingsWifi(View view) {
        WifiManager wifiManager = (WifiManager) getSystemService(Context.WIFI_SERVICE);
        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.Q) {
            wifiManager.setWifiEnabled(true);
            startActivity(new Intent(android.provider.Settings.ACTION_WIFI_SETTINGS));
        } else {
            ActivityResultLauncher.launch(new Intent(Settings.Panel.ACTION_WIFI));
        }
    }

    public void settingsMobileData(View view) {
        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.Q)startActivity(new Intent(Settings.ACTION_WIRELESS_SETTINGS));
        else ActivityResultLauncher.launch(new Intent(Settings.Panel.ACTION_INTERNET_CONNECTIVITY));
    }

    public void connected(View view) {
        finish();
        startActivity(new Intent(this, SplashScreen.class));
    }

    ActivityResultLauncher<Intent> ActivityResultLauncher = registerForActivityResult(
            new ActivityResultContracts.StartActivityForResult(),
            result -> {
                finish();
                startActivity(new Intent(this, SplashScreen.class));
            });
}

File: app/src/main/java/com/example/ibook_social_network/network/ConnectionsActivity.kt
package com.example.ibook_social_network.network

import android.content.Context
import android.content.Intent
import android.content.pm.ActivityInfo
import android.net.wifi.WifiManager
import android.os.Build
import android.os.Bundle
import android.provider.Settings
import android.view.View
import androidx.activity.result.contract.ActivityResultContracts
import androidx.appcompat.app.AppCompatActivity
import com.example.ibook_social_network.R
import com.example.ibook_social_network.loading.SplashActivity


class ConnectionsActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_connections)
        requestedOrientation = ActivityInfo.SCREEN_ORIENTATION_PORTRAIT
    }

    fun settingsWifi(view: View?) {
        val wifiManager = getSystemService(Context.WIFI_SERVICE) as WifiManager
        wifiManager.isWifiEnabled = true

        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.Q) {
            activityResultLauncher.launch(Intent(Settings.ACTION_WIFI_SETTINGS))
        }
        else{
            activityResultLauncher.launch(Intent(Settings.Panel.ACTION_WIFI))
        }
    }

    fun settingsMobileData(view: View?) {
        if (Build.VERSION.SDK_INT < Build.VERSION_CODES.Q)startActivity(Intent(Settings.ACTION_WIRELESS_SETTINGS))
        else activityResultLauncher.launch(Intent(Settings.Panel.ACTION_INTERNET_CONNECTIVITY))
    }

    fun connected(view: View?) {
        finish()
        startActivity(Intent(this, SplashActivity::class.java))
    }

    private val activityResultLauncher =
        registerForActivityResult(ActivityResultContracts.StartActivityForResult()) {
            finish()
            startActivity(Intent(this, SplashActivity::class.java))
        }
}